02_housing/2a_affordable_available_place.R

```diff
@@ -0,0 +1,264 @@
+# that income level or at a lower level or is vacant. 
+
+
+# Process:
+# (1) Housekeeping
+# (2) Import housing affordability measures calculated in 1a_housing_place.R
+# (3) At rent and down rent -  A unit is available at a given level of income if (1) it is affordable at that
+#     level, and (2) it is occupied by a renter either at that income level or at a lower level or is vacant. 
+#     (3a) Calculate the "at rent" number of units occupied at the appropriate income level or a lower level
+#         Do this for all units, and for renter and owner subgroups 
+#     (3b) Summarize at_rent for each subgroup and number of units/households at each
+#       income threshold by place
+# (4) calculate the number of affordable and available units at each income level per 100 households
+# (5) Create the Data Quality variable
+# (6) Clean and export
+#     (6a) subgroup file
+#     (6b) overall file
+# (7) Quality Check
+
+###################################################################
+
+# (1) Housekeeping
+# Set working directory to [gitfolder]: Open mobility-from-poverty.Rproj to make sure all file paths will work
+
+# Libraries you'll need
+library(tidyverse)
+library(tidylog)
+library(ipumsr)
+library(readxl)
+library(skimr)
+
+###################################################################
+
+# (2) Import housing affordability data (created in 1a_housing_place.R)
+
+# Either run "1a_housing_place.R" OR: Import the already prepared housing affordability and vacancy files 
+
+# this file is at the household level still so we can 
+# calculate the availability metric
+households_2022 <- read_csv("data/temp/households_2022.csv")
+
+# this file is at the place level for calculating the overall number
+# of affordable and available units per 100 households in step 4
+vacant_summed_2022 <- read_csv("data/temp/vacant_summed_2022.csv")
+
+###################################################################
+
+# (3) Availability -  A unit is available at a given level of income if (1) it is affordable at that
+#     level, and (2) it is occupied by a renter either at that income level or at a lower level or is vacant. 
+#     
+#    -  supply is number of housing units in each income bracket. 
+#         This is calculated in step 5 of the 1a_housing_place.R file (variables: Affordable80AMI_all, Affordable80AMI_renter, Affordable80AMI_owner)
+#    -  demand is number of households in each income bracket 
+#         This is calculated in step 5 of the 1a_housing_place.R file (variables: below80AMI, below50AMI, below30AMI)
+
+#    (3a) Calculate the "at rent" number of units occupied at the appropriate income level or a lower level
+#         Do this for all units, and for renter and owner subgroups 
+#         e.g., unit is affordable at 30 AMI and household is below 30 AMI and the unit is occupied (include vacancy in step 4)
+
+
+available_2022 <- households_2022 %>% 
```

Comment:
Since these are based on variables from the previous script that I think may have issues those issues flow through here.